src/auth/auth.service.ts

```diff
@@ -34,35 +34,56 @@ export class AuthService {
         }
     }
 
-    GetAllUsers(): Promise<User[]> {
-        const user = this.userRepository.find()
+    async GetAllUsers(page, size): Promise<any> {
+
+        const user = await this.userRepository.find({
+            take: size,
+            skip: size * (page - 1)
+        })
+
+        if (user.length === 0) {
+            throw new UnauthorizedException(`end of the page`)
+        }
+
         return user
     }
 
 
-    async searchByUsername(authCredDto: AuthCredDto, page): Promise<any> {
+    async searchByUsername(authCredDto: AuthCredDto, page, page_size): Promise<any> {
         const { user_name } = authCredDto
 
-        const user = await this.userRepository.find({ user_name })
+        const user = await this.userRepository.find(
+            {
+                where: {
+                    user_name: user_name,
+                },
+                take: page_size,
+                skip: page_size * (page - 1)
+            }
+        )
 
-        console.log(user.length)
+        // console.log(user)
+        if (user.length === 0) {
+            throw new UnauthorizedException(`End of The Page`)
```

Comment:
How will the front end know that the end of the page is reached to avoid this error ?

Reply:
if (user.length === 0) {
            throw new UnauthorizedException(`End of The Page`)
} 
i think this ll create a problem with frontend so i removed it.
i frontend they ll have a condition like user.length == 0 if true "End of the Page"